### ISSUE TITLE
Is Rails supported on Now?

### ISSUE THREAD

**Issue by antonnyman** [vercel/vercel#2998]
OS: `OSX 10.14.6`
Node: `10.16.0`
Ruby: `2.6.1`
Now CLI `16.1`
Rails: `6.0`

Hi,

I'm having issues running Rails locally using `now dev` and on `now`.

Locally with `now dev`
Reproduction steps:
1. `rails new rails-app`
2. Create a `now.json` with contents:
```json
{
  "version": 2,
  "builds": [
    {
      "src": "rails-app/config.ru",
      "use": "@now/ruby"
    }
  ],
  "routes": [{ "src": "/", "dest": "rails-app/config.ru" }]
}

```
3. `now dev` yields:
```sh
> Creating initial build

> Error! Command failed with exit code 2 (ENOENT): yum install -y git gcc make tar bzip2 readline-devel openssl-devel ruby-devel zlib-devel
spawn yum ENOENT
```

On `now`
1. `rails new rails-app`
2. Create a `now.json` with contents:
```json
{
  "version": 2,
  "builds": [
    {
      "src": "rails-app/config.ru",
      "use": "@now/ruby"
    }
  ],
  "routes": [{ "src": "/", "dest": "rails-app/config.ru" }]
}
```
3.  Now yields:
```sh
yarn info @now/build-utils@latest...
yarn info @now/ruby...
yarn add @now/build-utils@0.10.0...
yarn add v1.17.3
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @now/build-utils@0.10.0
info All dependencies
└─ @now/build-utils@0.10.0
Done in 0.56s.
yarn add @now/ruby@0.1.6...
yarn add v1.17.3
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @now/ruby@0.1.6
info All dependencies
└─ @now/ruby@0.1.6
Done in 1.10s.
TypeError: util.getSystemErrorName is not a function
    at getCode (/tmp/e97d8416e11caeb8/.build-utils/.builder/node_modules/@now/ruby/dist/index.js:28461:16)
    at makeError (/tmp/e97d8416e11caeb8/.build-utils/.builder/node_modules/@now/ruby/dist/index.js:28500:35)
    at handlePromise (/tmp/e97d8416e11caeb8/.build-utils/.builder/node_modules/@now/ruby/dist/index.js:3759:26)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
worker exited with code 20 and signal null
done
done
```

Deployment URL: rails-64wuh4wd8.now.sh

Any idea what this could be?

**Comment by styfle** [comment]
Hi @antonnyman 

Ruby doesn't support `now dev` yet.

I'll add this as a feature request, thanks!

----

Update: I just realized the was an error in the now deployment. We will investigate the `TypeError: util.getSystemErrorName is not a function`, thanks

**Comment by styfle** [comment]
Hi @antonnyman 

Upon further investigation, the error is likely due to Ruby 2.6 which adds additional features that are not available in Ruby 2.5.

Can you try deploying Rails with Ruby 2.5? Thanks!

https://zeit.co/docs/v2/advanced/builders#ruby-version

**Comment by rauchg** [comment]
@antonnyman while this is in theory could work, we recommend you follow a different app structure. Try to make your frontend static, and use API functions written in Ruby.
